### ISSUE TITLE
UARegistrationDelegate methods not behaving as expected

### ISSUE THREAD

**Issue by petarbelokonski** [urbanairship/ios-library#131]
On iOS 10, in my AppDelegate, I do :

```
UAConfig *config = [UAConfig defaultConfig];
[UAirship takeOff:config]; 
[UAirship push].registrationDelegate = self;
[[UAirship push] setUserPushNotificationsEnabled:YES];
```

When I tap on 'Allow' on the system dialog the `notificationRegistrationFinished` method is called with the default options. However when I tap on 'Don't allow' the same method gets called with the same options ? What am I missing here ? I would like to get the same behaviour as the `application:didRegisterUserNotificationSettings `callback for iOS 9 that I was using like this:

```
  func application(_ application: UIApplication, didRegister notificationSettings: UIUserNotificationSettings) {
        if notificationSettings.types.isEmpty {
            print("user tapped don't allow button")
        } else {
            print("user tapped allow button ")
        }
    }
```
. Many thanks.

**Comment by rlepinski** [comment]
@petarbelokonski We only call that method after we determine we need to register for APNS. If its already up to date we skip registering again. That probably explains why you are only getting the call once.

You are right that we are not passing back the correct values. We will do a patch to address that issue in the next few days.

**Comment by petarbelokonski** [comment]
@rlepinski I believe in the `updateAPNSRegistration` method of UAPush.m the `options` parameter that gets passed to `notificationRegistrationFinishedWithOptions:categories:` is not the right one. Looking forward to your fix. Best

**Comment by rlepinski** [comment]
@petarbelokonski Just released 8.3.3. `notificationRegistrationFinishedWithOptions:categories:` will not be called after the user accepts permissions on iOS 8&9 and with the authorized options.
